### ISSUE TITLE
Systemd-enabled distro fails to boot with unhelpful error message if systemd is not installed

### ISSUE THREAD

**Issue by reynoldsa** [microsoft/WSL#8838]
### Version

0.67.6.0

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.62.1-microsoft-standard-WSL2

### Distro Version

Custom Fedora

### Repro Steps

Enable systemd via `wsl.conf` in a distro that does not have `systemd` installed.

### Expected Behavior

Launch fails, with an error message indicating why it didn't work (e.g. something about systemd being unavailable).

### Actual Behavior

Terminal message says `Catastrophic Failure` with error code `Wsl/Service/0x8000ffff`.

**Comment by benhillis** [comment]
Fair - could / should probably do a quick check if the binary actually exists. I'll make that change.

**Comment by benhillis** [comment]
FYI for others that put their distro into a bad state by doing this, you can edit your /etc/wsl.conf file by:

%userprofile%\\.wslconfig
```
[wsl2]
safeMode=true
```

**Comment by reynoldsa** [comment]
> Got the same issue with custom Fedora 36 after addendum to `/etc/wsl.conf` with:
> 
> ```ini
> [boot]
> systemd=true
> ```

@bricss Try `safeMode` as above, then `sudo dnf install systemd -y`. Shutdown WSL after with `wsl --shutdown`, unset safe mode, then try again.

**Comment by bricss** [comment]
@reynoldsa thank you! works like a charm ✨

**Comment by CsehRafin** [comment]
> @reynoldsa thank you! works like a charm ✨

Also Btw Why this error happens like when i turn on systemd by editing /etc/wsl.conf 
when upgrading ubuntu 
error :
Reading package lists... Done
Building dependency tree
Reading state information... Done
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 libnss-systemd : Depends: systemd (= 249.11-0ubuntu3.6) but 245.4-4ubuntu3.18 is installed
 libpam-systemd : Depends: systemd (= 249.11-0ubuntu3.6) but 245.4-4ubuntu3.18 is installed
 systemd : Depends: libsystemd0 (= 245.4-4ubuntu3.18) but 249.11-0ubuntu3.6 is installed
 systemd-sysv : Depends: systemd (= 249.11-0ubuntu3.6) but 245.4-4ubuntu3.18 is installed
 systemd-timesyncd : Depends: systemd (= 249.11-0ubuntu3.6) but 245.4-4ubuntu3.18 is installed
 udev : Breaks: systemd (< 249.11-0ubuntu3.6) but 245.4-4ubuntu3.18 is installed
        Recommends: systemd-hwe-hwdb but it is not installed

**Comment by paller** [comment]
```
WSL --version
WSL version 1.0.0.0.0
```
I did the same as @bricss but can't enable safe mode on Windows11 WSLg version 1. I enabled the debug console in .wslconfig and that works. The kernel is doing something before it fails, but I'm unable to get a shell such that I can disable or install systemd.

**Comment by somu1795** [comment]
> ```
> WSL --version
> WSL version 1.0.0.0.0
> ```
> 
> I did the same as @bricss but can't enable safe mode on Windows11 WSLg version 1. I enabled the debug console in .wslconfig and that works. The kernel is doing something before it fails, but I'm unable to get a shell such that I can disable or install systemd.

Same issue here
WSL version: 1.0.0.0
Kernel version: 5.15.74.2
WSLg version: 1.0.47
MSRDC version: 1.2.3575
Direct3D version: 1.606.4
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.900

Using fedora in wsl.
Adding safe mode doesn't work , however I can go into default ubuntu distro

**Comment by paller** [comment]
@somu1795 you can work around it by converting it to the old WSL version, not to be confused with WSL release version, if that makes sense.
```
wsl --set-version <Distro> 1
```
It took me a few minutes before it was done, but after that it was able to run. I could then delete the wsl.conf file and convert it back to WSL 2.

**Comment by lucivskvn** [comment]
@paller Thank you for the workaround, it's working on my case [openSUSE-Leap-15.4]

**Comment by marcoaureliocardoso** [comment]
Mounting the vhdx image file on another distro made possible to edit the /etc/wsl.conf on faulty system.

**On Windows:**
`wsl -d <working system> --mount --vhd <path to image>\<image name>.vhdx --bare`
Example: wsl -d Ubuntu-20.04 --mount --vhd c:\wsl\vms\Ubuntu-22.04\Ubuntu-22.04.vhdx --bare

**Inside the working system:**
Mount the device on a mount point (Example: sudo mount /dev/sdd /mnt)
Edit the \<mount point\>/etc/wsl.conf, removing the **[boot] systemd=true** part.
Umount and exit the system.

**Back on Windows, umount the image:**
`wsl --unmount \\?\<path to image>\<image name>.vhdx`
Example: wsl --unmount \\?\c:\wsl\vms\Ubuntu-22.04\Ubuntu-22.04.vhdx

Check if the former faulty system boot up.

**Comment by SamB** [comment]
I *knew* it was a good idea to add a "make sure `systemd` is installed" step to the documentation. 

Unfortunately, since I've only installed one distro under WSL and it was so long ago that I don't remember if I uninstalled systemd myself or if it started out like that, I wasn't sure what to say about whether it's likely to be preinstalled etc. and so didn't feel qualified to add that step to the documentation myself.

And I definitely hadn't thought about the fact that it would be difficult to fix things if you tried to boot a distro with systemd enabled but not installed, so I wouldn't have added a "warning" block or whatever to call out the danger even if I *had* written this up and it had gotten merged. Oh well.
